spark-cassandra-connector/src/main/scala/org/apache/spark/metrics/CassandraConnectorSource.scala

```diff
@@ -37,10 +38,7 @@ class CassandraConnectorSource extends Source {
 }
 
 object CassandraConnectorSource {
-  @volatile
-  private var _instance: Option[CassandraConnectorSource] = None
-  @volatile
-  private var _env: SparkEnv = null
+  private var _state: (Option[CassandraConnectorSource], SparkEnv) = (None, null)
```

Comment:
You can use `Option.apply` to convert nulls / non-nulls to `None` / `Some` automatically.